requests/adapters.py

```diff
@@ -493,13 +493,12 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                 # TODO: Remove this in 3.0.0: see #2811
                 if not isinstance(e.reason, NewConnectionError):
                     raise ConnectTimeout(e, request=request)
-
-            if isinstance(e.reason, ResponseError):
+            elif isinstance(e.reason, ResponseError):
```

Comment:
I was probably (incorrectly) trying to catch only one exception for each of these, rather than multiple.